pimcore/models/Tool/CustomReport/Config.php

```diff
@@ -150,7 +150,23 @@ public static function getReportsList()
     public static function getAdapter($configuration, $fullConfig = null)
     {
         $type = $configuration->type ? ucfirst($configuration->type) : 'Sql';
-        $adapter = "\\Pimcore\\Model\\Tool\\CustomReport\\Adapter\\{$type}";
+        if ($type === 'Custom') {
+            if (@\class_exists($configuration->className, true)) { // can need autoloader, can send warning and break json response so we use @
+                $adapter = $configuration->className;
+                $adpaterInstance = new $adapter($configuration, $fullConfig);
+                // we make some check for security reason (because className can be choosen at the GUI level)
+                if (get_parent_class($adpaterInstance) !== 'Pimcore\\Model\\Tool\\CustomReport\\Adapter\\AbstractAdapter') {
```

Comment:
Thanks, @Yivan, I will change it just after an internal discussion about the sense of that change. 
For me, it's a nice feature but I'm not sure, maybe it would be better idea to make configuration PHP file which is responsible for listing additional adapters. In an array, you can specify additional adapters. For example:

`"Microsoft AX orders" => "\Website\Report\Ax"`

I'm not a big fun of solutions which allow the administration panel users specifying the configuration based on the code of the application.